File: semestr-8/Jezyk C#/Wyklad-3/SamochodyLibrary/DataStructures/Addons/LeatherSeats.cs
using SamochodyLibrary.DataStructures.Cars;

namespace SamochodyLibrary.DataStructures.Addons;

public class LeatherSeats : CarDecorator
{
    public LeatherSeats(Car car) : base(car)
    {
    }

    public override string name()
    {
        return "Leather seats";
    }

    public override string info()
    {
        return base.info() + " + Leather seats";
    }

    public override decimal price()
    {
        return base.price() + 2000M;
    }
}

File: semestr-8/Programowanie obiektowe w Java/alexander/POJ/Archiwum/2018s/Decorator/src/LeatherSeats.java
public class LeatherSeats extends CarDecorator
{
   public LeatherSeats(Car car) { super(car); }
   public @Override String info() { return super.info()+" + Skórzana tapicerka"; }
   public @Override double price() { return super.price()+15000; }
}
